infra/src/dsl/xroad.ts

```diff
@@ -178,6 +183,19 @@ export const Education = new XroadConf({
 
 export const NationalRegistry = new XroadConf({
   env: {
+    XROAD_NATIONAL_REGISTRY_SERVICE_PATH: {
+      dev: 'IS-DEV/GOV/10001/SKRA-Protected/Einstaklingar-v1',
+      staging: 'IS-TEST/GOV/6503760649/SKRA-Protected/Einstaklingar-v1',
+      prod: 'IS/GOV/6503760649/SKRA-Protected/Einstaklingar-v1',
+    },
+    // Only cache on dev for now.
+    XROAD_NATIONAL_REGISTRY_REDIS_NODES: {
+      dev:
+        '["clustercfg.general-redis-cluster-group.5fzau3.euw1.cache.amazonaws.com:6379"]',
```

Comment:
Could this be a comma separated list instead?

Reply:
No ... 😆 

The new [config library](https://github.com/island-is/island.is/tree/main/libs/nest/config) has native support for parsing and validating JSON environment variables. IMO it is more precise, flexible and has [better DX](https://github.com/island-is/island.is/blob/7dbef7d5134b832458a9653fed45fd68c53f82e7/libs/clients/national-registry/v2/src/lib/nationalRegistryClient.config.ts) for some types of configuration.

It would be cool to upgrade the dsl to support this better:

```
XROAD_NATIONAL_REGISTRY_REDIS_NODES: {
      dev: json([
        'clustercfg.general-redis-cluster-group.5fzau3.euw1.cache.amazonaws.com:6379',
      ]),
},
```

Happy to jump on a call to argue about this. God knows there's plenty of stuff in these [two](https://github.com/island-is/island.is/pull/5612) PRs worth chatting about. 😃

Reply:
It's also much more error prone. This'll get translated to regular ENV variables in our deployments. In the case of an emergency (example: we need to provision a new redis cluster and point all deployments to that) the first step would be to manually update the deployments, then to fix it in the dsl and wait until it builds etc.

I'd be much more comfortable editing a comma separated list than a json encoded array :grimacing:

Reply:
I have absolute faith in the JSON writing skills of the devops team. 🤓 Worst case the config validation stops the deployment and tells you what you did wrong.

I've also seen comma separated environment variables messed up many times, with whitespace or buggy empty/undefined handling. I also see devs often adding more elaborate syntax and stuff to these variables to support extra metadata.